mmtrack/datasets/base_sot_dataset.py

```diff
@@ -0,0 +1,285 @@
+from torch.utils.data import Dataset
+
+from mmtrack.core.evaluation import eval_sot_ope
+from mmtrack.datasets import DATASETS
+
+
+@DATASETS.register_module()
+class SOTDataset(Dataset, metaclass=ABCMeta):
+    """Dataset of single object tracking.
+
+    The dataset can both support training and testing mode.
+    """
+
+    CLASSES = None
+
+    def __init__(self,
+                 img_prefix,
+                 pipeline,
+                 split,
+                 test_mode=False,
+                 bbox_min_size=0,
+                 load_as_video=True,
+                 **kwargs):
+        self.img_prefix = img_prefix
+        self.split = split
+        self.pipeline = Compose(pipeline)
+        self.test_mode = test_mode
+        self.bbox_min_size = bbox_min_size
+        # for DistributedVideoSampler in testing
+        self.laod_as_video = load_as_video
+        ''' The self.data_info is a list, which the length is the
+            number of videos. The default content is in the following format:
+            [
+                {
+                    'video_path': the video path
+                    'ann_path': the annotation path
+                    'start_frame_id': the starting frame number contained in
+                                    the image name
+                    'end_frame_id': the ending frame number contained in the
+                                    image name
+                    'framename_template': the number of digits in the image
+                                    name
+                },
+                ...
+            ]
+        '''
+        self.data_infos = self.load_data_infos(split=self.split)
+        self.num_frames_per_video = [
+            self.get_len_per_video(video_ind)
+            for video_ind in range(len(self.data_infos))
+        ]
+
+    def __getitem__(self, ind):
+        if self.test_mode:
+            assert isinstance(ind, tuple)
+            # the first element in the tuple is the video index and the second
+            # element in the tuple is the frame index
+            return self.prepare_test_img(ind[0], ind[1])
+        else:
+            return self.prepare_train_img(ind)
+
+    @abstractmethod
+    def load_data_infos(self, split='train'):
+        pass
+
+    def get_img_names_from_video(self, video_ind):
+        """Get all frame paths in a video.
+
+        Args:
+            video_ind (int): video index
+
+        Returns:
+            list[str]: all image paths
+        """
+        img_names = []
+        start_frame_id = self.data_infos[video_ind]['start_frame_id']
+        end_frame_id = self.data_infos[video_ind]['end_frame_id']
+        framename_template = self.data_infos[video_ind]['framename_template']
+        for frame_id in range(start_frame_id, end_frame_id + 1):
+            img_names.append(
+                osp.join(
+                    self.data_infos[video_ind]['video_path'],
+                    '{frame_id:0{framename_template}}.jpg'.format(
```

Comment:
The `framename_template` shoud not be an int. We can set `framename_template = '%06d.jpg'` in a specifical dataset , and use `framename_template % frame_id` to generate the filename. You can give it a try.

Reply:
It doesn't work. May be we can rename `framename_template` as 'num_digit_in_framename'?